source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.cc

```diff
@@ -112,6 +118,7 @@ void GetAddrInfoDnsResolver::resolveThreadRoutine() {
 
   while (true) {
     PendingQuerySharedPtr next_query;
+    absl::optional<uint32_t> num_retries = absl::nullopt;
```

Comment:
nit: out of curiosity is the explicit initialization needed, or does the default constructor do the right thing? No need to change it, I'm just curious.

Reply:
It does. Sometimes I prefer being explicit, but I can remove it. Done.